### PR TITLE
[stable/spring-cloud-data-flow] Update to new SCDF version 2.4.0

### PR DESCRIPTION
**Pull request by chrisjs**
#### What this PR does / why we need it:

- update scdf version to 2.4.0
- update skipper version to 2.3.0
- update chart dependency versions

Signed-off-by: Chris Schaefer <cschaefer@pivotal.io>

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/spring-cloud-data-flow/README.md [hunk 1]
```diff
@@ -146,10 +146,11 @@ The following tables list the configurable parameters and their default values.
...
 ### RabbitMQ Configuration
 
-| Parameter                  | Description                              | Default                   |
-| -------------------------- | ---------------------------------------- | ------------------------- |
-| rabbitmq.enabled           | Enable RabbitMQ as the middleware to use | true
-| rabbitmq.rabbitmqUsername  | RabbitMQ user name                       | user
+| Parameter                   | Description                                 | Default                   |
+| --------------------------- | ------------------------------------------- | ------------------------- |
+| rabbitmq.enabled            | Enable RabbitMQ as the middleware to use    | true
+| rabbitmq.rabbitmq.username  | RabbitMQ user name                          | user
```
Comment (sabbyanandan) [inline stable/spring-cloud-data-flow/README.md:152]: Seeing `rabbitmq` twice looks odd. Could we maybe change it to just `rabbitmq.username` and `rabbitmq.password` or is it a breaking change?

Comment (chrisjs) [reply stable/spring-cloud-data-flow/README.md:152]: unfortunately that wart came as part of upgrading... looks like some props were moved down a level (under `rabbitmq` block), so in the rabbit yaml itself its for example `rabbitmq.username`, we have a leading `rabbitmq` as thats how chart refs are done

**Review by sabbyanandan — APPROVED** [review]
/lgtm

**Pull request merged**